remix-analyzer/src/solidity-analyzer/index.ts

```diff
@@ -1,25 +1,31 @@
 'use strict'
 import { AstWalker } from 'remix-astwalker'
 import list from './modules/list'
+import { CompilationResult, AnalyzerModule, AnalysisReportObj, AnalysisReport } from 'types'
+
+type ModuleObj = {
+  name: string
+  mod: AnalyzerModule
+}
 
 export default class staticAnalysisRunner {
 
-  run (compilationResult, toRun, callback) {
-    const modules = toRun.map((i) => {
-      const m = this.modules()[i]
+  run (compilationResult: CompilationResult, toRun: any[], callback: ((reports: AnalysisReport[]) => void)): void {
+    const modules: ModuleObj[] = toRun.map((i) => {
+      const m: AnalyzerModule = this.modules()[i]
       return { 'name': m.name, 'mod': m }
     })
-
     this.runWithModuleList(compilationResult, modules, callback)
   }
 
-  runWithModuleList (compilationResult, modules, callback) {
-    let reports: any[] = []
+  runWithModuleList (compilationResult: CompilationResult, modules: ModuleObj[], callback: ((reports: AnalysisReport[]) => void)): void {
+    let reports: AnalysisReport[] = []
     // Also provide convenience analysis via the AST walker.
-    const walker = new AstWalker()
+    const walker: AstWalker = new AstWalker()
```

Comment:
Normally you shouldn't have to hard type it here. But doing `const walker = new AstWalker()` typescript should know the type.

Reply:
ok